hyperspy/misc/slicing.py

```diff
@@ -186,9 +186,14 @@ def _get_array_slices(self, slices, isNavigation=None):
                 array_slices.append(slice_)
         return array_slices
 
-    def _slicer(self, slices, isNavigation=None):
+    def _slicer(self, slices, isNavigation=None, out=None):
         array_slices = self._get_array_slices(slices, isNavigation)
-        _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        if out is None:
+            _obj = self._deepcopy_with_new_data(self.data[array_slices])
+        else:
+            _obj = out
+            _obj.axes_manager = self.axes_manager.deepcopy()
+            _obj.data = self._data[array_slices]  # Changes data id!
```

Comment:
@francisco-dlp, will you redo it in this way? Given the main point of the PR, I agree with @vidartf, that retaining axes is important.

I could not find `Axis.update_from` in master or in this code, so you'll have to add (and test) that as well, if you go this way